### DISCUSSION TITLE
Creating custom script and combine IntegerVar with Dropdown

### DISCUSSION THREAD

**Discussion by Netjamcoder** [netbox-community/netbox#18228] [Help Wanted!]
Hello everyone,

i try to create a custom script with a field like commit_rate or port_speed which looks like:
![netbox2](https://github.com/user-attachments/assets/f24fa5a4-00c4-4e57-91c8-76becfc68727)

commitrate = IntegerVar(
   label = "Garantierte Bandbreite (Kbps)",
   description="Garantierte Bandbreite"
   required=false
)

It should be possible to type in an integer value or to select a value from a drop down menu.

Any ideas?

**Comment by candlerb** [comment]
https://netboxlabs.com/docs/netbox/en/stable/customization/custom-scripts/#choicevar

**Reply by Netjamcoder** [reply]
It's not what i need. Only Integer values should be allowed and it should be also possible to type in a value or to choose a value like in my screenshot.

**Reply by candlerb** [reply]
If you want only to choose between pre-defined values, then it will do what you want. If you want to enter an arbitrary value, then you need to have an integer value field.

I don't think there's any field which has a mixture of both behaviours - i.e. a dropdown menu which can be overridden to give arbitrary values. You'd have to have two separate fields in that case, and combine them in your custom script.
